src/zope/untrustedpython/interpreter.py

```diff
@@ -47,15 +45,17 @@ def eval(self, globals, locals=None):
 
 
 class CompiledProgram(object):
-    """A compiled expression
-    """
+    """A compiled program."""
 
     def __init__(self, source, filename='<string>'):
         self.source = source
-        self.code = compile(source, filename, 'exec')
+        self.code = rcompile.compile(source, filename, 'exec')
 
     def exec_(self, globals, locals=None, output=None):
         globals['__builtins__'] = SafeBuiltins
+        globals['_print_'] = PrintCollector
```

Comment:
Hm, print was a statement but is now a function, so eval() now can also print, so I wonder if CompiledExpression needs to do something about `_print_`?

Also I wonder why the global `exec_code` didn't do anything about printing.

Maybe they don't need to.

Reply:
`CompiledExpression` is not used anywhere throughout whole `zopefoundation`. So it could be a feature request if someone has a use for storing the output in `CompiledExpression` like `CompiledProgram` does.

`exec_code` is also not used outside this repository. So maybe no-one uses it.